ogr/abstract.py

```diff
@@ -1708,6 +1708,14 @@ def get_web_url(self) -> str:
         """
         raise NotImplementedError()
 
+    # TODO: Consider ‹get_sha_from(branch=None, tag=None)›
+    def get_sha_from_branch(self, branch: str) -> Optional[str]:
```

Comment:
I'd say having two separate methods for each is a better design :) `get_sha_from` sounds like it would be too complicated and prone to errors - also with modern editors you are immediately being offered a list methods so it's easier to pick one which has narrow scope